src/snowflake/snowpark/mock/analyzer.py

```diff
@@ -172,6 +173,10 @@ def analyze(
             so that it will detect column name change.
             however, in the result calculation, we want to column name to be the output name, which is 'totB',
             so we set keep_alias to False in the execution.
+            quote_literal: if true, the literal string will be returned single quoted as '<literal>', column name
```

Comment:
Please correct me if I'm wrong: does `quote_literal` apply to any expressions that contain a literal value?

Reply:
for now, `quote_literal=True` is only applied to the right child of `Like` and `RegExp` in my code change

Reply:
updated to use `calculate_expression` to literal computation and single quote literal all the time in analyze